src/Kernel32.Tests/storebanned/Kernel32Facts.cs

```diff
@@ -944,6 +944,184 @@ public void SetHandleInformation_DoesNotThrow()
             HandleFlags.HANDLE_FLAG_NONE));
     }
 
+    /// <summary>
+    /// Basic validation for <see cref="Kernel32.CreateThread(SECURITY_ATTRIBUTES*, SIZE_T, THREAD_START_ROUTINE, IntPtr, CreateProcessFlags, int*)"/>
+    ///
+    /// Creates a thread by supplying <see cref="CreateThread_Test_ThreadMain(IntPtr)"/> as its ThreadProc/<see cref="Kernel32.THREAD_START_ROUTINE"/>.
+    /// The ThreadProc updates a bool value (supplied by the thread that created it) from false -> true. This change
+    /// is observed by the calling thread as proof of successful thread-creation.
+    ///
+    /// Also validates that the (native) Thread-ID for the newly created Thread is different than the (native) Thread-ID
+    /// of that of the calling thread.
+    /// </summary>
+    [Fact]
+    public unsafe void CreateThread_Test()
+    {
+        var secAttrs = new Kernel32.SECURITY_ATTRIBUTES
+        {
+            bInheritHandle = 1,
+            lpSecurityDescriptor = IntPtr.Zero,
+            nLength = Marshal.SizeOf<Kernel32.SECURITY_ATTRIBUTES>()
```

Comment:
```suggestion
            nLength = sizeof(Kernel32.SECURITY_ATTRIBUTES)
```
I know that this is just a test code, but it is a few orders of magnide faster and more correct to use `sizeof` here.

Reply:
(same in other places)

Reply:
Interesting. Can you teach me more about when `sizeof` is OK to use for structs for interop purposes? When must `Marshal.SizeOf` be used? We use `Marshal.SizeOf` in many places in this repo so if `sizeof` is acceptable we can switch to it everywhere.

Reply:
`Marshal.SizeOf` returns the size of the unmanaged view of the struct. `sizeof` returns the size of the managed view of the struct.

Unmanaged and managed views are identical for the blittable structs. It is better to use `sizeof` instead of `Marshal.SizeOf` for blittable structs. They return the same number and `sizeof` is much cheaper.

The code in this method is passing the managed view of the struct into the PInvoke without any marshaling. Unmanaged pointers like `SECURITY_ATTRIBUTES*` in this case are never marshalled. It is why I have said that it is more correct to use `sizeof` here.

Reply:
Awesome. *all* of our structs in this repo are blittable, and our guidelines reinforce this. So it sounds like we should go through all of our use of `Marshal.SizeOf` and replace with `sizeof`. I'll open an issue to track.